File: weekly-contest/144/corporate-flight-bookings.js
const { expect } = require('chai');

/**
 * @param {number[][]} bookings
 * @param {number} n
 * @return {number[]}
 */
let corpFlightBookings = function(bookings, n) {
  const result = [];
  for (let i = 0; i <= n; i++) result[i] = 0;

  for (let booking of bookings) {
    result[booking[0] - 1] += booking[2];
    result[booking[1]] -= booking[2];
  }

  for (let i = 1; i < n; i++) {
    result[i] += result[i - 1];
  }
  return result.slice(0, n);
};

it('corporate-flight-bookings', () => {
  expect(corpFlightBookings([[1,2,10],[2,3,20],[2,5,25]], 5)).to.deep.eq([10, 55, 45, 25, 25]);
});

File: weekly-contest/144/corporate-flight-bookings.go
package main

import (
	"fmt"
)

func corpFlightBookings(bookings [][]int, n int) []int {
	result := make([]int, n+1)

	for _, booking := range bookings {
		result[booking[0]-1] += booking[2]
		result[booking[1]] -= booking[2]
	}

	for i := 1; i < n; i++ {
		result[i] += result[i-1]
	}

	return result[0:n]
}

func main() {
	fmt.Println(corpFlightBookings(
		[][]int{
			[]int{1, 2, 10},
			[]int{2, 3, 20},
			[]int{2, 5, 25},
		}, 5,
	))
}
